auth/src/android/credential_android.cc

```diff
@@ -358,13 +366,21 @@ Credential::~Credential() {
 
 Credential::Credential(const Credential& rhs)
     : impl_(nullptr), error_code_(kAuthErrorNone) {
-  *this = rhs;
+  if (rhs.impl_ != nullptr) {
+    JNIEnv* env = GetJniEnv();
+    jobject j_cred_ref = env->NewGlobalRef(static_cast<jobject>(rhs.impl_));
+    impl_ = static_cast<void*>(j_cred_ref);
+  } else {
+    impl_ = nullptr;
+  }
+  error_code_ = rhs.error_code_;
+  error_message_ = rhs.error_message_;
```

Comment:
This is pretty much the same implementation to `operator=()`, isn't it?
Why not just use the original `*this = rhs;`